index.js

```diff
@@ -533,12 +539,14 @@ function parse (args, opts) {
     var type = 'boolean'
 
     if (flags.strings && flags.strings[key]) type = 'string'
+    else if (flags.numbers && flags.numbers[key]) type = 'number'
     else if (flags.arrays && flags.arrays[key]) type = 'array'
 
     return type
   }
 
   function isNumber (x) {
+    if (flags.numbers && flags.numbers[x] && !isNaN(x)) return true
```

Comment:
I think you can get rid of this check, it won't actually do anything, instead update:

https://github.com/yargs/yargs-parser/blob/master/index.js#L315

To be something like:

```js
var value = val
if (!checkAllAliases(key, flags.strings)) {
  if (isNumber(val) || checkAllAliases(key, flags.numbers)) value = Number(val)
}
```